shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java

```diff
@@ -0,0 +1,313 @@
+              break;
+            }
+          case SplitInstallSessionStatus.CANCELED:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install canceled.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              sessionIdToName.remove(state.sessionId());
+              break;
+            }
+          case SplitInstallSessionStatus.CANCELING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install canceling.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.PENDING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install pending.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.REQUIRES_USER_CONFIRMATION:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install requires user confirmation.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.DOWNLOADING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) downloading.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.DOWNLOADED:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) downloaded.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.INSTALLING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) installing.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          default:
+            Log.d(TAG, "Status: " + state.status());
+        }
+      }
+    }
+  }
+
+  public PlayStoreDynamicFeatureManager(@NonNull Context context, @NonNull FlutterJNI flutterJNI) {
+    this.context = context;
+    this.flutterJNI = flutterJNI;
+    splitInstallManager = SplitInstallManagerFactory.create(context);
+    listener = new FeatureInstallStateUpdatedListener();
+    splitInstallManager.registerListener(listener);
```

Comment:
this seems a bit suspicious. I would expect the destroy call on the engine to pass through to a destroy call on the interface as well. And an implementation here should unregister everything.